src/ssl.c

```diff
@@ -46581,6 +46745,36 @@ static WC_INLINE int SCSV_Check(byte suite0, byte suite)
     return 0;
 }
 
+static WC_INLINE int sslCipherMinMaxCheck(const WOLFSSL *ssl, byte suite0,
+        byte suite)
+{
+    const CipherSuiteInfo* cipher_names = GetCipherNames();
+    int cipherSz = GetCipherNamesSize();
+    int i;
+    for (i = 0; i < cipherSz; i++)
+        if (cipher_names[i].cipherSuite0 == suite0 &&
+                cipher_names[i].cipherSuite == suite)
+            break;
+    if (i == cipherSz)
+        return 1;
+    if (cipher_names[i].minor < ssl->options.minDowngrade)
+        return 1;
+    switch (cipher_names[i].minor) {
```

Comment:
TLS 1.0 cipher suites are available in TLS 1.1 and 1.2 but not 1.3.
Does this code need to handle these cases?

Reply:
This is only used in `wolfSSL_get_ciphers_compat` so it is used to return the available ciphersuites to the user. I changed this to:
```
    if (cipher_names[i].minor < ssl->options.minDowngrade) {
        if (ssl->options.minDowngrade <= TLSv1_2_MINOR &&
                cipher_names[i].minor >= TLSv1_MINOR)
            /* 1.0 ciphersuites are in general available in 1.1 and
             * 1.1 ciphersuites are in general available in 1.2 */
            return 0;
        return 1;
    }
```
This will show ciphers as available if the min version is <= 1.2